sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ContextualSensor.java

```diff
@@ -0,0 +1,104 @@
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugin.typescript;
+
+import com.google.gson.Gson;
+import com.google.gson.stream.JsonReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import org.sonar.api.batch.InstantiationStrategy;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.rule.CheckFactory;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.plugin.typescript.SensorContextUtils.ContextualAnalysisRequest;
+import org.sonarsource.api.sonarlint.SonarLintSide;
+
+import static org.sonar.plugin.typescript.SensorContextUtils.getInputFiles;
+
+@InstantiationStrategy(InstantiationStrategy.PER_BATCH)
+@SonarLintSide
+public class ContextualSensor implements Sensor {
+
+  private static final Logger LOG = Loggers.get(ContextualSensor.class);
+
+  private CheckFactory checkFactory;
+  private static AtomicReference<Socket> socketReference = new AtomicReference<>();
+
+  public ContextualSensor(CheckFactory checkFactory) {
+    this.checkFactory = checkFactory;
+  }
+
+  @Override
+  public void describe(SensorDescriptor sensorDescriptor) {
+    sensorDescriptor.onlyOnLanguage(TypeScriptLanguage.KEY).name("Contextual SonarTS").onlyOnFileType(InputFile.Type.MAIN);
+  }
+
+  @Override
+  public void execute(SensorContext sensorContext) {
+    Iterable<InputFile> inputFiles = getInputFiles(sensorContext);
+    inputFiles.forEach(inputFile ->
+      connect().ifPresent(socket -> {
+        try {
+          final OutputStreamWriter writer = new OutputStreamWriter(socket.getOutputStream(), StandardCharsets.UTF_8);
+          LOG.error("ANALYZING " + inputFile.contents()); // TODO remove this
+          TypeScriptRules typeScriptRules = new TypeScriptRules(checkFactory);
+          String request = getContextualRequest(inputFile, typeScriptRules);
+          LOG.info("REQUEST " + request); // TODO remove this
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=sonarlint-integration-poc&issues=AWDgL5jKbBlXo5JPxg62&open=AWDgL5jKbBlXo5JPxg62)

[ISSUE_KEY]: AWDgL5jKbBlXo5JPxg62